File: FrontEnd/app/src/main/java/byui/app/linkUp/LoginActivity.java
package byui.app.linkUp;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import java.util.HashMap;
import java.util.Map;

import byui.app.linkUp.FrontendSender.FrontendSender;
import byui.app.linkUp.databinding.LoginBinding;

public class LoginActivity extends AppCompatActivity {

  private LoginBinding binding;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    binding = LoginBinding.inflate(getLayoutInflater());
    View view = binding.getRoot();
    setContentView(view);

    setupLoginButton();
    setupSignUpButton();
  }

  private void setupLoginButton() {
    binding.signinButton.setOnClickListener(v -> {
      String username = binding.UserNEntry.getText().toString().trim();
      String password = binding.passEntry.getText().toString().trim();

      new LoginTask().execute(username, password);
    });
  }

  private void setupSignUpButton() {
    binding.tosignup.setOnClickListener(v -> {
      Intent intent = new Intent(LoginActivity.this, SignupActivity.class);
      startActivity(intent);
    });
  }

  private class LoginTask extends AsyncTask<String, Void, Boolean> {

    @Override
    protected Boolean doInBackground(String... params) {
      try {
        String username = params[0];
        String password = params[1];
        String hashedPassword = HashUtil.hashPassword(password);

        Map<String, String> data = new HashMap<>();
        data.put("username", username);
        data.put("password_hash", hashedPassword);

        FrontendSender.sendDataToBackend("userLogin", data);
      } catch (Exception e) {
        e.printStackTrace();
        return false;
      }
      return true;
    }

    @Override
    protected void onPostExecute(Boolean success) {
      if (success) {
        Toast.makeText(LoginActivity.this, "Login Successful", Toast.LENGTH_SHORT).show();
        Intent intent = new Intent(LoginActivity.this, AccountActivity.class);
        startActivity(intent);
        finish();
      } else {
        TextView signin_error = findViewById(R.id.signin_error);
        signin_error.setVisibility(View.VISIBLE);
        Toast.makeText(LoginActivity.this, "Login Failed", Toast.LENGTH_SHORT).show();
      }
    }
  }
}

File: FrontEnd/app/src/main/java/byui/app/linkUp/LoginActivity.kt
package byui.app.linkUp

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import byui.app.linkUp.databinding.LoginBinding

class LoginActivity : AppCompatActivity() {

    // Using View Binding to access the layout's views
    private lateinit var binding: LoginBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = LoginBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Setup listeners for your buttons
        setupLoginButton()
//        setupSignUpButton()
    }

    private fun setupLoginButton() {
        binding.signinButton.setOnClickListener {
            // Implement your login logic here
            // For example, validate the username and password
            val username = binding.UserNEntry.text.toString().trim()
            val password = binding.passEntry.text.toString().trim()

//            // Assume a method validateCredentials exists and returns true if the credentials are valid
//            if (validateCredentials(username, password)) {
//                // Start MainActivity on successful login
//                val intent = Intent(this, MainActivity::class.java)
//                startActivity(intent)
//                finish() // Finish LoginActivity so it's removed from the back stack
//            } else {
//                // Show error message
//            }
        }
    }

//    private fun setupSignUpButton() {
//        binding.tosignup.setOnClickListener {
//            // Navigate to SignUpActivity
//            // Assume SignUpActivity is another activity you have for users to sign up
//            val intent = Intent(this, SignUpActivity::class.java)
//            startActivity(intent)
//        }
//    }

    private fun validateCredentials(username: String, password: String): Boolean {
        // Implement your validation logic here
        // This is just a placeholder for demonstration
        return true
    }
}